packages/x-data-grid-pro/src/typeOverloads/modules.ts

```diff
@@ -42,8 +43,20 @@ export interface GridEventLookupPro {
   rowOrderChange: { params: GridRowOrderChangeParams };
   /**
    * Fired when a new batch of rows is requested to be loaded. Called with a [[GridFetchRowsParams]] object.
+   * Used to trigger `onFetchRows`.
    */
   fetchRows: { params: GridFetchRowsParams };
+  // Data source
+  /**
+   * Fired to make a new request through the data source's `getRows` method.
+   * @ignore - do not document.
+   */
+  getRows: { params: GridGetRowsParams };
+  /**
+   * Fired when the data request is resolved either via the data source or from the cache.
+   * @ignore - do not document.
+   */
+  rowsFetched: {};
```

Comment:
https://github.com/mui/mui-x/pull/13878#discussion_r1797344577

cc @arminmeh

Reply:
Events removed, since https://github.com/mui/mui-x/pull/13878/commits/6ef5bf6023456a83cf6a55fc8ed6d4a768ac7277

Reply:
The strategy processing looks like a very solid way to handle this 👍